Doc/c-api/type.rst

```diff
@@ -185,6 +185,13 @@ Type Objects
 
    .. versionadded:: 3.11
 
+.. c:function:: PyObject* PyType_GetFullyQualName(PyTypeObject *type)
```

Comment:
Why can't it just be GetFullyQualifiedName?

Reply:
We already have a `PyType_GetQualName()` function which returns `type.__qualname__`: https://docs.python.org/dev/c-api/type.html#c.PyType_GetQualName

I followed the `PyType_GetQualName()` name: `PyType_GetFullyQualName()` returns `type.__fullyqualname__`.

Reply:
I know that, but it still tickles my grammar bone. :-(

Reply:
Ok, I renamed the function to `PyType_GetFullyQualifiedName()`.

I prefer to keep ``__fullyqualname__`` for the attribute name. We already have ``__qualname__``. And since it's commonly used, I prefer to make it short.

An alternative is to use FQN acronym :-D

* `type.__fqn__`
* `PyType_GetFQN()`

But I prefer "longer" names, so I don't need to check the doc to know what they mean ;-)

On the Internet, Domain Name Service (DNS) use [FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name): Fully Qualified Domain Name :-)

Reply:
I share Guido's unease with "fullyqualname": it sounds wrong to have the abbreviated "qual" between two unabbreviated words.

For comparison:
- Several dunders, including some of the most recently added ones, include an underscore in the word: `__class_getitem__`, `__release_buffer__`, `__type_params__`, `__init_subclass__`, `__text_signature__`
- Other dunders include multiple words, often abbreviated, without underscores: `__getnewargs__`, `__kwdefaults__`, `__isabstractmethod__`

I would prefer `__fully_qualified_name__`. It's on the long side, but it matches the current name for the concept (mentioned in https://docs.python.org/3.12/glossary.html#term-qualified-name). `__fqn__` seems too cryptic.

Reply:
I concur with Jelle.